### PR TITLE
Modified Binary Search

### LINKED ISSUE #1224: Updating files to take input at runtime (#1160)
Updating these files : 
Binary_Search.cpp
Binary_Search.java
Binary_Search.go
Binary_Search.py
searching_a_circularly_sorted_array.cpp

### PR DESCRIPTION
**Pull request by singh-shreya6**
Fixes #1224 referencing #1160 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Changes proposed in this pull request:
- Added dynamic inputs and example

#### Languages Used:
- C++
- Java
- Python
- Go

#### Files Added:
- Binary_Search.cpp
- Binary_Search.java
- Binary_Search.py
- Binary_Search.go
- searching_a_circularly_sorted_array.cpp

### CODE REVIEW

File: Binary_Search/Binary_Search.cpp [hunk 1]
```diff
@@ -62,42 +62,56 @@ int Upper_Bound(int array[], int size, int desired)
...
-Not Found
 4
 6
 6
 0
 
+Input:
+num = 5
+array = {1, 3, 5, 7, 9}
+desired  = 2
```
Comment (SakshiShreya) [inline Binary_Search/Binary_Search.cpp]: Remove the extra space before `=`

File: Binary_Search/Binary_Search.go [hunk 2]
```diff
@@ -28,11 +28,40 @@ func BinarySearch(array []int, desired int) (index int) {
 }
 
 func main() {
-	array := []int{1, 2, 3, 4, 5, 6, 7}
-	index := BinarySearch(array, 4)
+	var num int
```
Comment (SakshiShreya) [inline Binary_Search/Binary_Search.go]: Convert all tabs to 4 spaces.

File: Binary_Search/Binary_Search.java [hunk 3]
```diff
@@ -23,25 +25,35 @@ else if (desired < array[middle])
...
+desired = 4
 
-Found
-Not Found
+Output = Found
 
+Input:
+num = 5
+array = {1, 3, 5, 7, 9}
+desired  = 2
```
Comment (SakshiShreya) [inline Binary_Search/Binary_Search.java]: Remove the extra space before `=`

File: Binary_Search/Binary_Search.py [hunk 4]
```diff
@@ -1,33 +1,44 @@
 # Function for Binary Search
-def Binary_Search(list, desired):
+def binary_Search(array, desired):
```
Comment (SakshiShreya) [inline Binary_Search/Binary_Search.py]: Rename the function to camel case

File: Binary_Search/Binary_Search.java [hunk 5]
```diff
@@ -23,25 +25,35 @@ else if (desired < array[middle])
...
+        if (binarySearch(array, num, desired) != -1)
             System.out.println("Found");
         else
             System.out.println("Not Found");
     }
 }
 
-/* Output
+/* 
+Input :
```
Comment (SakshiShreya) [inline Binary_Search/Binary_Search.java]: Remove space before `:`

File: Binary_Search/Binary_Search.go [hunk 6]
```diff
@@ -28,11 +28,40 @@ func BinarySearch(array []int, desired int) (index int) {
...
 	if index != -1 {
-		fmt.Println("Found 4 at ", strconv.Itoa(index))
+		fmt.Println("Found at ", strconv.Itoa(index))
 	} else {
 		fmt.Println("Not Found")
 	}
 }
+
+/* Output
+Input :
```
Comment (SakshiShreya) [inline Binary_Search/Binary_Search.go]: Remove space before `:`

File: Binary_Search/Binary_Search.cpp [hunk 7]
```diff
@@ -62,42 +62,56 @@ int Upper_Bound(int array[], int size, int desired)
...
     // Element greater than 0
-    cout << Upper_Bound(array, 7, 0) << endl;
+    cout << upperBound(array, num, 0) << endl;
 
     return 0;
 }
 
-/* Output
+/* 
+Input :
```
Comment (SakshiShreya) [inline Binary_Search/Binary_Search.cpp]: Remove space before `:`

File: Binary_Search/Binary_Search.py [hunk 8]
```diff
@@ -1,33 +1,44 @@
...
-# Element to be searched is 4
-print ("Found") if Binary_Search(list, 4) != -1 else ("Not Found")
+desired = int(input())
 
-# Element to be searched is 9
-print ("Found") if Binary_Search(list, 9) != -1 else ("Not Found")
+print ("Found") if binary_Search(array, desired) != -1 else ("Not Found")
 
+'''
+Input :
```
Comment (SakshiShreya) [inline Binary_Search/Binary_Search.py]: Remove space before `:`

File: Binary_Search/searching_a_circularly_sorted_array.cpp [hunk 9]
```diff
@@ -28,7 +28,31 @@ int searchCircular(int arr[], int len, int num)
...
+        cin >> arr[i];
+    } 
+    int desired;
+    cin >> desired;
+    cout << searchCircular(arr, num, desired); // Returns the index of element
     return 0;
 }
+
+/*
+Input :
```
Comment (SakshiShreya) [inline Binary_Search/searching_a_circularly_sorted_array.cpp]: Remove space before `:`

**Comment by singh-shreya6** [comment]
@SakshiShreya Done. Please Review.

File: Binary_Search/searching_a_circularly_sorted_array.cpp (hunk 9 above)
Comment (singh-shreya6) [reply Binary_Search/searching_a_circularly_sorted_array.cpp]: Made all the requested changes @SakshiShreya

**Pull request merged**